barte/models.py

```diff
@@ -0,0 +1,118 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional, List, Dict, Any
+
+@dataclass
+class Customer:
+    name: str
+    tax_id: str
+    email: str
+
+@dataclass
+class CardToken:
+    id: str
+    type: str
+    created_at: datetime
+    last_digits: str
+    holder_name: str
+    expiration_month: int
+    expiration_year: int
+    brand: str
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "CardToken":
+        if isinstance(data["created_at"], str):
+            data["created_at"] = datetime.fromisoformat(data["created_at"].replace("Z", "+00:00"))
+        return cls(**data)
+
+@dataclass
+class Charge:
+    id: str
+    amount: int
+    currency: str
+    status: str
+    payment_method: str
+    description: Optional[str]
+    customer: Customer
+    created_at: datetime
+    metadata: Optional[Dict[str, Any]] = None
+    installments: Optional[int] = None
+    installment_amount: Optional[int] = None
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "Charge":
+        # Copy data to avoid modifying the original
+        data = data.copy()
+        
+        # Convert customer if it's a dict
+        if isinstance(data["customer"], dict):
+            data["customer"] = Customer(**data["customer"])
+        
+        # Convert created_at if it's a string
+        if isinstance(data["created_at"], str):
+            data["created_at"] = datetime.fromisoformat(data["created_at"].replace("Z", "+00:00"))
+        
+        return cls(**data)
```

Comment:
[dacite](https://github.com/konradhalas/dacite)?

> This module simplifies creation of data classes from dictionaries.

Reply:
I didn't know about it, thanks